code/datums/components/rust.dm

```diff
@@ -19,22 +20,25 @@
 	var/atom/wall_new = new /turf/open/floor/plating(src)
 	wall_new.AddComponent(/datum/component/rust)
 
-/datum/component/rust/Initialize(rust_iconstate = "rust")
+/datum/component/rust/Initialize(iconstate = "rust", icon) // Maybe swap these args?
 	. = ..()
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
 	var/atom/parent_atom = parent
-	if(!(rust_iconstate in icon_states(parent_atom.icon)))
+	icon = icon ? icon : parent_atom.icon // If an icon is supplied use that, if it doesnt use the parent atoms icon
+	if(!(iconstate in icon_states(icon)))
 		return COMPONENT_INCOMPATIBLE
-
-	rust_overlay = mutable_appearance(parent_atom.icon, rust_iconstate)
+	rust_overlay = mutable_appearance(icon, iconstate)
 
 /datum/component/rust/RegisterWithParent()
 	RegisterSignal(parent, COMSIG_ATOM_UPDATE_OVERLAYS, .proc/apply_rust_overlay)
 	RegisterSignal(parent, list(COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_RUSTSCRAPER)), .proc/secondary_tool_act)
 	RegisterSignal(parent, COMSIG_PARENT_PREQDELETED, .proc/parent_del)
 	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/handle_examine)
 	ADD_TRAIT(parent, TRAIT_RUSTY, src)
+	// Unfortunately registering with parent sometimes doesn't cause an overlay update
+	var/atom/parent_atom = parent
+	parent_atom.update_icon(UPDATE_OVERLAYS)
```

Comment:
Put this in `Initialize` (as well as that ADD_TRAIT). As far as I know, RegisterWithParent should only be for signals.